predicators/approaches/grammar_search_invention_approach.py

```diff
@@ -341,7 +341,8 @@ def _quantified_types(self) -> List[Type]:
         ]
 
     def _classify_object(self, s: State, obj: Object) -> bool:
-        assert obj.type == self.body.types[self.free_variable_idx]
+        # assert obj.type == self.body.types[self.free_variable_idx]
+        assert obj.is_instance(self.body.types[self.free_variable_idx])
```

Comment:
The previous code does not support object type inheritance because it checks via `==`. Checking via `is_instance()` also checks if `obj`'s parent type matches. 

Do you want me to ^ write this explanation as a comment in the code? I think it's unnecessary.

Reply:
nope - that's fine.